python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1781,29 +1790,156 @@ def set_requires_typed_event_stream(self, *, error_message: Optional[str] = None
         self.op_execution_context.set_requires_typed_event_stream(error_message=error_message)
 
 
+class AssetCheckExecutionContext:
+    def __init__(self, op_execution_context: OpExecutionContext) -> None:
+        self._op_execution_context = check.inst_param(
+            op_execution_context, "op_execution_context", OpExecutionContext
+        )
+        self._step_execution_context = self._op_execution_context._step_execution_context  # noqa: SLF001
+
+    @staticmethod
+    def get() -> "AssetExecutionContext":
```

Comment:
maybe I'm just missing something small, but why is this method returning an `AssetExecutionContext` instead of an `AssetCheckExecutionContext`? The error message on line 1807 is also a bit confusing to me

Reply:
good catch!